config/src/token_remaps.rs

```diff
@@ -0,0 +1,158 @@
+//! Defines helpers for fetching or parsing a token remap
+//!
+//! See https://github.com/renegade-fi/token-mappings/tree/main for more information
+
+use arbitrum_client::constants::Chain;
+use bimap::BiMap;
+use common::types::token::TOKEN_REMAPS;
+use serde::{Deserialize, Serialize};
+use util::raw_err_str;
+
+/// The base URL for raw token remap files
+const REMAP_BASE_URL: &str = "https://raw.githubusercontent.com/renegade-fi/token-mappings/main/";
+
+// --------------------
+// | Serialized Types |
+// --------------------
+
+/// The token remap type
+///
+/// Contains a series of token info objects
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+struct TokenRemap {
+    /// The token information in the remap file
+    tokens: Vec<TokenInfo>,
+}
+
+/// The token info type
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+struct TokenInfo {
+    /// The name of the token
+    name: String,
+    /// The token's ticker
+    ticker: String,
+    /// The address of the token in the chain
+    address: String,
+    /// The number of decimals the token uses in the ERC20 representation
+    decimals: u8,
+}
+
+impl TokenRemap {
+    /// Convert the token mapping into a map of token addresses to ticker
+    pub fn to_remap(&self) -> BiMap<String, String> {
+        self.tokens.iter().map(|info| (info.address.clone(), info.ticker.clone())).collect()
+    }
+}
+
+/// Setup token remaps in the global `OnceCell`
+pub fn setup_token_remaps(remap_file: Option<String>, chain: Chain) -> Result<(), String> {
+    // If the remap file is not provided, fetch the Renegade maintained remap file
+    // from the default location
+    let map = if let Some(file) = remap_file {
+        parse_remap_from_file(file)
+    } else {
+        fetch_remap_from_repo(chain)
+    }?;
+
+    // Update the static token remap with the given one
+    let remap = map.to_remap();
+    TOKEN_REMAPS.set(remap).map_err(raw_err_str!("Failed to set token remap: {:?}"))
+}
+
+/// Parse a token remap from a JSON file
+fn parse_remap_from_file(file_path: String) -> Result<TokenRemap, String> {
+    // Read the file into a string
+    let file = std::fs::read_to_string(file_path)
+        .map_err(raw_err_str!("Failed to read remap file: {}"))?;
+    serde_json::from_str(&file).map_err(raw_err_str!("Failed to parse remap from file: {}"))
+}
+
+/// Pull the token remap from the repo
+fn fetch_remap_from_repo(chain: Chain) -> Result<TokenRemap, String> {
+    let url = format!("{}{}.json", REMAP_BASE_URL, chain);
+    let resp = reqwest::blocking::get(&url)
```

Comment:
nit: clippy saying to:
```suggestion
    let resp = reqwest::blocking::get(url)
```